Sources/SimplenoteSearch/Extensions/NSPredicate+Search.swift

```diff
@@ -74,30 +77,39 @@ extension NSPredicate {
         return NSPredicate(format: "tags MATCHES[n] %@", regex)
     }
 
-    /// Returns a NSPredicate that will match Tags with a given Keyword
+    /// Returns a NSPredicate that will match Tags with a given Query
     ///
     /// -   We'll always analyze the last token in the string
     ///     -   Whenever the last token contains `tag:`, we'll lookup Tags with names containing the payload, excluding exact matches
     ///     -   Otherwise we'll match Tags **containing** the last token
     ///     -   If the `tag:` operator has no actual payload, **every single tag** will be matched (Always True Predicate)
     ///
     @objc
-    public static func predicateForTag(keyword: String) -> NSPredicate {
-        let keywords = keyword.trimmingCharacters(in: .whitespacesAndNewlines).components(separatedBy: .whitespaces)
-        let last = keywords.last?.lowercased() ?? String()
-
-        guard let tag = last.suffix(afterPrefix: lowercasedSearchOperatorForTags) else {
-            return NSPredicate(format: "name CONTAINS[cd] %@", last)
+    public static func predicateForTag(query: SearchQuery) -> NSPredicate {
+        guard let lastItem = query.items.last else {
```

Comment:
There's something a bit confusing about the API signature. Only the last `SearchQueryItem` is evaluated, but the method may get multiple items.

This API perhaps could be private, and expect a single `SearchQueryItem`, assumed to be `.tag` maybe?

Reply:
This method is used from `NotesListController` / `NoteListState` to show tags related to the current search, we can't make it private.

I think it's okay to pass the whole query and let this method decide which part of the query it should use to show relevant tags. Maybe we can rename it to `predicateForTags` instead of `predicateForTag`.

Right now it it works like this:
* `... tag:test`: shows tags that contain `test`
* `... test`: shows tags that contain `test`
* `... tag:`: shows all tags

The last case is also why we need to keep empty tags around.

Reply:
Renamed the method to `predicateForTags(in query: SearchQuery) -> NSPredicate`
Thanks for the suggestions!